core/src/main/java/com/linecorp/armeria/client/DefaultHttpClient.java

```diff
@@ -46,27 +52,40 @@ public HttpResponse execute(HttpRequest req) {
     }
 
     private HttpResponse execute(@Nullable EventLoop eventLoop, HttpRequest req) {
-        final URI uri = URI.create(req.path());
-        final String authority = uri.getAuthority();
+        URI uri;
 
-        if (isUndefinedUri(uri()) && authority == null) {
-            return HttpResponse.ofFailure(new IllegalArgumentException("no authority: " + req.path()));
-        } else if (authority != null) {
-            final Endpoint endpoint = Endpoint.parse(authority);
-            final RequestHeaders newHeaders = req.headers().toBuilder().path(uri.getRawPath()).build();
-            final HttpRequest newReq = HttpRequest.of(req, newHeaders);
-            return execute(eventLoop, endpoint, newReq);
+        if (isAbsolutePath(req.path())) {
+            try {
+                uri = URI.create(req.path());
+            } catch (Exception ex) {
+                logger.warn("Failed to create URI: {}", req.path(), ex);
+                uri = null;
+            }
         } else {
-            final String originalPath = req.path();
-            final String newPath = concatPaths(uri().getRawPath(), originalPath);
-            final HttpRequest newReq;
-            // newPath and originalPath should be the same reference if uri().getRawPath() can be ignorable
-            if (newPath != originalPath) {
-                newReq = HttpRequest.of(req, req.headers().toBuilder().path(newPath).build());
+            uri = null;
+        }
+
+        if (uri == null) {
+            if (isUndefinedUri(uri())) {
+                req.abort();
+                return HttpResponse.ofFailure(new IllegalArgumentException("no authority: " + req.path()));
             } else {
-                newReq = req;
+                final String originalPath = req.path();
+                final String newPath = concatPaths(uri().getRawPath(), originalPath);
+                final HttpRequest newReq;
+                // newPath and originalPath should be the same reference if uri().getRawPath() can be ignorable
+                if (newPath != originalPath) {
+                    newReq = HttpRequest.of(req, req.headers().toBuilder().path(newPath).build());
+                } else {
+                    newReq = req;
+                }
+                return execute(eventLoop, endpoint(), newReq);
             }
-            return execute(eventLoop, endpoint(), newReq);
+        } else {
+            final Endpoint endpoint = Endpoint.parse(uri.getAuthority());
+            final RequestHeaders newHeaders = req.headers().toBuilder().path(uri.getRawPath()).build();
+            final HttpRequest newReq = HttpRequest.of(req, newHeaders);
+            return execute(eventLoop, endpoint, newReq);
```

Comment:
How about inverting the `if` block and unwrapping the `else` block? i.e.

```java
if (uri != null) {
    ...
    return execute(eventLoop, endpoint, newReq);
}

if (isUndefinedUri(uri())) {
    ...
    return ...;
}

final String originalPath = req.path();
...
return execute(...);
```

This way, we get less indentation and cognitive load.